wai-aria/scripts/aria-utils.js

```diff
@@ -156,6 +156,35 @@ const AriaUtils = {
     }
   },
 
+  verifyLevelsBySelector: function (selector) {
+    const els = document.querySelectorAll(selector);
+    if (!els.length) {
+      throw `Selector passed in verifyLevelsBySelector("${selector}") should match at least one element.`;
+    }
+    for (const el of els) {
+      let testName = el.getAttribute("data-testname");
+      promise_test(async t => {
+        const expectedLevel = el.getAttribute("data-expectedlevel");
+        // TODO: Yuck. This is surely wrong.
+        // Look for ariaLevel first, otherwise grab only the digit(s) from
+        // the nodeName (e.g., `H2` --> 2).
+        const ariaLevel = el.ariaLevel || el.nodeName.replace(/\D/g, '')
+        assert_equals(ariaLevel, expectedLevel, el.outerHTML);
+      }, `${testName}`);
+    }
+  },
 
+  verifyNoHeadingAncestor: function (selector) {
+    const els = document.querySelectorAll(selector);
+    if (!els.length) {
+      throw `Selector passed in verifyNoHeadingAncestor("${selector}") should match at least one element.`;
+    }
+    for (const el of els) {
+      let testName = el.getAttribute("data-testname");
+      promise_test(async t => {
```

Comment:
Does this need to be an async promise test? Seems like it could be a standard test since it's not waiting for a WebDriver callback.

Reply:
Actually it does need to be a promise test, because it will need to walk up the ancestor change getting `computedRole` from WebDriver, rather than using the DOM-local `:is()` query.

And, if self and an ancestor return `heading` as the `computedRole`, output should include that, so the ARIA WG can look at the results and attempt to determine: "Which is _right_?"